src/coinbase/address/external_address.ts

```diff
@@ -217,33 +232,132 @@ export class ExternalAddress extends Address {
    */
   private async getStakingBalances(
     assetId: string,
-    options: CoinbaseExternalAddressStakeOptions,
+    options: StakingContextOptions = { mode: StakeOptionsMode.DEFAULT },
   ): Promise<{ [key: string]: Decimal }> {
-    const request = {
-      network_id: this.getNetworkId(),
-      asset_id: Asset.primaryDenomination(assetId),
-      address_id: this.getId(),
-      options: this.transformStakeOptions(options),
-    };
+    let typedContext: StakingContextContext;
+
+    if (this.isEthereumPartialStakingContextOptions(assetId, options)) {
+      const request = {
+        network_id: this.getNetworkId(),
+        asset_id: Asset.primaryDenomination(assetId),
+        address_id: this.getId(),
+        options: this.transformStakingContextOptions(options, assetId),
+      };
+
+      const response = await Coinbase.apiClients.stake!.getStakingContext(request);
+
+      typedContext = response!.data.context as PartialEthStakingContext;
+
+      return {
+        stakeableBalance: Balance.fromModelAndAssetId(typedContext.stakeable_balance, assetId)
+          .amount,
+        unstakeableBalance: Balance.fromModelAndAssetId(typedContext.unstakeable_balance, assetId)
+          .amount,
+        claimableBalance: Balance.fromModelAndAssetId(typedContext.claimable_balance, assetId)
+          .amount,
+      };
+    } else if (this.isEthereumNativeStakingContextOptions(assetId, options)) {
+      const nativeEthStakingOptions: { [key: string]: string } = {};
 
-    const response = await Coinbase.apiClients.stake!.getStakingContext(request);
+      // TODO(rohit) cdp-service backend should work without this.
+      nativeEthStakingOptions["validator_pub_keys"] =
+        "0x8000011bc03bbf99ac5964d14d3bb52de983c848cc3734d736235a19715e8cbbd5e963163eb4bd2d8cd473d103b95c12";
+
+      nativeEthStakingOptions["mode"] = StakeOptionsMode.NATIVE;
+
+      const request = {
+        network_id: this.getNetworkId(),
+        asset_id: Asset.primaryDenomination(assetId),
+        address_id: this.getId(),
+        options: nativeEthStakingOptions,
+      };
+
+      const response = await Coinbase.apiClients.stake!.getStakingContext(request);
+
+      typedContext = response!.data.context as NativeEthStakingContext;
+
+      let unstakeableBalance = new Decimal(0);
+
+      typedContext.validators.forEach(validator => {
+        unstakeableBalance = unstakeableBalance.add(new Decimal(validator.balance.amount));
+      });
```

Comment:
we should move this logic into the server

Reply:
yup done 👍